src/doc/man/cargo-test.md

```diff
@@ -27,20 +27,34 @@ on 3 threads in parallel:
 
     cargo test foo -- --test-threads 3
 
-Tests are built with the `--test` option to `rustc` which creates an
-executable with a `main` function that automatically runs all functions
-annotated with the `#[test]` attribute in multiple threads. `#[bench]`
-annotated functions will also be run with one iteration to verify that they
-are functional.
+Tests are built with the `--test` option to `rustc` which creates a special
+executable by linking your code with libtest. The executable automatically
+runs all functions annotated with the `#[test]` attribute in multiple threads.
+`#[bench]` annotated functions will also be run with one iteration to verify
+that they are functional.
+
+If the package contains multiple test targets, each target compiles to a
+special executable as aforementioned, and then is run serially.
 
 The libtest harness may be disabled by setting `harness = false` in the target
 manifest settings, in which case your code will need to provide its own `main`
 function to handle running tests.
 
+### Documentation tests
+
 Documentation tests are also run by default, which is handled by `rustdoc`. It
-extracts code samples from documentation comments and executes them. See the
-[rustdoc book](https://doc.rust-lang.org/rustdoc/) for more information on
-writing doc tests.
+extracts code samples from documentation comments of the library target, and
+then executes them.
+
+Different from normal test targets, each code block compiles to a doctest
+executable on the fly with `rustc`. These executables runs in parallel in
+separate processes. The compilation of a code block is in fact a part of test
+function controlled by libtest, so some options such as `--jobs` might not
+take effect. Note that this execution model of doctests is an undocumented
+implementation details, beware of depending on it.
```

Comment:
This reads a little strange to me since this is documentation specifying how it works. Perhaps something like this?

```suggestion
take effect. Note that this execution model of doctests is not guaranteed
and may change in the future; beware of depending on it.
```